src/analyze/analyze.c

```diff
@@ -1832,6 +2010,50 @@ static int do_security(int argc, char *argv[], void *userdata) {
         return analyze_security(bus, strv_skip(argv, 1), 0);
 }
 
+static int parse_arg_color_style(DotArgColorStyle color_or_style, const char *arg_dot_color_style) {
+        char delim[] = ":";
+        const char *p;
+        int r;
+
+        p = arg_dot_color_style;
+        _cleanup_free_ char *word = NULL;
+
+        r = extract_first_word(&p, &word, delim, 0);
+        if (r == 0)
+                return r;
+        if (r == -ENOMEM)
+                return log_oom();
+        if (r < 0) {
+                log_error("Invalid syntax, ignoring: %s", arg_dot_color_style);
+                return r;
+        }
+
+        for (UnitDependency dependency = UNIT_REQUIRES; dependency <= UNIT_TRIGGERS; ++dependency) {
+                if (streq(unit_dependency_to_string(dependency), word)) {  /* unit found */
+                        r = extract_first_word(&p, &word, delim, 0);
```

Comment:
no need to call extract_first_word() again. You don#t expect more than one ":" specifier, do you? Hence, just use `p` after you parsed the first word off...

Reply:
also you are leaking the memory allocated by he first word here

Reply:
done